src/Faker/Provider/fa_IR/Person.php

```diff
@@ -134,4 +134,82 @@ class Person extends \Faker\Provider\Person
 
     protected static $titleMale = array('آقای', 'استاد', 'دکتر', 'مهندس');
     protected static $titleFemale = array('خانم', 'استاد', 'دکتر', 'مهندس');
+
+    /**
+     * this method returns a valid Iranian nationalCode
+     * @return string
+     * @example '8075859741'
+     */
+    public static function nationalCode()
+    {
+        $area = self::createAreaCode();
+        $core = self::createCoreCode();
+        $control = self::createControlCode($area, $core);
+
+        return sprintf("%03d%06d%01d", $area, $core, $control);
+    }
+
+    /**
+     * @return int|string
+     * this method generates a 3-digit valid area code to be used in nationalCode
+     */
+    private static function createAreaCode()
+    {
+        $area = 0;
+
+        while (self::sumOfDigits($area) == 0) {
+            $area = mt_rand(0, 9) . mt_rand(0, 9) . mt_rand(0, 9);
+        }
+
+        return $area;
+    }
+
+    /**
+     * @param $number
+     * @return int
+     * this helper method sums up the digits in a number
```

Comment:
useless comment, please remove

Reply:
You got it. Actually, since I updated the while loop which used this method, this method is now obsolete, so I omitted it.